build.gradle

```diff
@@ -105,6 +105,8 @@ allprojects {
 subprojects {
   apply plugin: 'eclipse'
   apply plugin: 'project-report'
+  apply plugin: 'java'
```

Comment:
2 issues here:
1. If this is needed, then I believe you can remove the `apply plugin: 'java'` line from each of the projects
2. I don't know much about this, but since the scala plugin extends the java plugin, could this affect the projects that depend on scala (particularly those in which we force joint compilation)?

Reply:
Turns out that line is not needed since each subproject already has the java or scala plugin. Removed in the latest commit.